snapcraft/__init__.py

```diff
@@ -222,6 +223,20 @@ def build(self):
             ignore=lambda d, s: common.SNAPCRAFT_FILES
             if d is self.sourcedir else [])
 
+    def clean_build(self):
```

Comment:
we had the concept on clean build (building in lxc) before this, which makes it confusing.
what about clean_build_step?